### ISSUE TITLE
Can't compile structs with the same name in different rust modules

### ISSUE THREAD

**Issue by OlegTheCat** [rustwasm/wasm-bindgen#2247]
### Describe the Bug

Structs with the same names, that reside in different rust modules and have `wasm_bindgen` attribute applied, cannot be compiled. If I understand correctly this is because `wasm_bindgen` generates JS bindings into a single JS file and doesn't follow rust modules structure.

### Steps to Reproduce

1. Create file `my_mod.rs` with the following content:

```
use wasm_bindgen::prelude::*;

#[wasm_bindgen]
pub struct MyStruct {
    x: usize
}
```

2. Add the following code to `lib.rs`:
```
use wasm_bindgen::prelude::*;
mod my_mod;

#[wasm_bindgen]
pub struct MyStruct {
    foo: String
}

```
3. Execute `wasm-pack build`.

### Expected Behavior

Rust code is successfully compiled.

### Actual Behavior

Compilation fails with `symbol `__wbg_mystruct_free` is already defined` error.

**Comment by alexcrichton** [comment]
Yes this is an unfortunate limitation of wasm-bindgen because there's no namespacing of exported types. They're simply all exported into the same namespace in JS, so there's no way to export the same name multiple times.

**Comment by OlegTheCat** [comment]
@alexcrichton, I see, thanks for the explanation. Is it something that is going to be addressed in the future? If no, I think, we can close this issue.

**Comment by alexcrichton** [comment]
At this time I don't have any plans to fix this myself, but if someone else is interested in fixing it I can help review! 

(I don't know how best to fix it right now)

**Comment by hugopeixoto** [comment]
As a workaround to avoid having to rename the rust structs, one could use `#[wasm_bindgen(js_name = MyModMyStruct)]` or something like that to avoid the name collision, right?

**Comment by iacore** [comment]
> As a workaround to avoid having to rename the rust structs, one could use `#[wasm_bindgen(js_name = MyModMyStruct)]` or something like that to avoid the name collision, right?

not working
